### PR TITLE
Better Vote Revealer

### PR DESCRIPTION
**Pull request by blackMasterLord**

### CODE REVIEW

File: Osiris/Hacks/Misc.cpp [hunk 1]
```diff
@@ -1088,6 +1089,44 @@ void Misc::voteRevealer(GameEvent& event) noexcept
...
+        const auto entity = interfaces->entityList->getEntity(ent_idx);
+        const auto isLocal = localPlayer && entity == localPlayer.get();
+
+        memory->clientMode->getHudChat()->printf(0, " \x0C\u2022Osiris\u2022 %c%s\x01 call vote (\x06%s\x01)", isLocal ? '\x01' : '\x06', isLocal ? "You" : entity->getPlayerName().c_str(), voteName(vote_type));
+    }
+}
+
+void Misc::onVotePass() noexcept
+{
+    memory->clientMode->getHudChat()->printf(0, " \x0C\u2022Osiris\u2022\x01 vote\x06 PASS");
```
Comment (danielkrupinski) [inline Osiris/Hacks/Misc.cpp:1122]: There should be a check if vote revealer is enabled.

File: Osiris/Hacks/Misc.cpp [hunk 2]
```diff
@@ -1088,6 +1089,44 @@ void Misc::voteRevealer(GameEvent& event) noexcept
...
+}
+
+void Misc::onVoteFailed() noexcept
+{
+    memory->clientMode->getHudChat()->printf(0, " \x0C\u2022Osiris\u2022\x01 vote\x07 FAILED");
```
Comment (danielkrupinski) [inline Osiris/Hacks/Misc.cpp:1127]: There should be a check if vote revealer is enabled.

**Comment by HowToMakeFaceitAB** [comment]
> Build errors need to be fixed.

hey?

File: Osiris/Hacks/Misc.cpp [hunk 3]
```diff
@@ -1080,6 +1081,50 @@ void Misc::voteRevealer(GameEvent& event) noexcept
...
+        {
+        case 0: return "Kick";
+        case 1: return "Change Level";
+        case 6: return "Surrender";
+        case 13: return "Start TimeOut";
+        default: return "ERROR";
+        }
+    };
+
+    ProtobufReader msg{ static_cast<const std::uint8_t*>(data), size };
```
Comment (danielkrupinski) [inline Osiris/Hacks/Misc.cpp:1100]: reader can be const

File: Osiris/Hacks/Misc.cpp [hunk 4]
```diff
@@ -1080,6 +1081,50 @@ void Misc::voteRevealer(GameEvent& event) noexcept
     memory->clientMode->getHudChat()->printf(0, " \x0C\u2022Osiris\u2022 %c%s\x01 voted %c%s\x01", isLocal ? '\x01' : color, isLocal ? "You" : entity->getPlayerName().c_str(), color, votedYes ? "Yes" : "No");
 }
 
+void Misc::onVoteStart(const void*& data, int& size) noexcept
```
Comment (danielkrupinski) [inline Osiris/Hacks/Misc.cpp:1084]: `data` and `size` arguments shouldn't be references

File: Osiris/Hacks/Misc.cpp [hunk 5]
```diff
@@ -1080,6 +1081,50 @@ void Misc::voteRevealer(GameEvent& event) noexcept
...
+        case 13: return "Start TimeOut";
+        default: return "ERROR";
+        }
+    };
+
+    ProtobufReader msg{ static_cast<const std::uint8_t*>(data), size };
+    const auto ent_idx = msg.readInt32(2);
+    const auto vote_type = msg.readInt32(3);
+
+    if (ent_idx && vote_type) {
```
Comment (danielkrupinski) [inline Osiris/Hacks/Misc.cpp:1104]: This check eliminates `vote_type` being `0` so `voteName` lambda will never return "Kick".

File: Osiris/Hacks/Misc.cpp [hunk 6]
```diff
@@ -1081,7 +1081,7 @@ void Misc::voteRevealer(GameEvent& event) noexcept
     memory->clientMode->getHudChat()->printf(0, " \x0C\u2022Osiris\u2022 %c%s\x01 voted %c%s\x01", isLocal ? '\x01' : color, isLocal ? "You" : entity->getPlayerName().c_str(), color, votedYes ? "Yes" : "No");
 }
 
-void Misc::onVoteStart(const void*& data, int& size) noexcept
+void Misc::onVoteStart(const void* data, int size) noexcept
```
Comment (floydGG) [inline Osiris/Hacks/Misc.cpp:1084]: got this error

image

**Review by danielkrupinski — APPROVED** [review]

**Comment by danielkrupinski** [comment]
Thanks!

**Pull request merged**